src/dmd/attrib.d

```diff
@@ -451,6 +451,66 @@ extern (C++) final class CPPMangleDeclaration : AttribDeclaration
     }
 }
 
+/***********************************************************
+ */
+extern (C++) final class CPPNamespaceDeclaration : AttribDeclaration
+{
+    Expression exp;
+
+    extern (D) this(Identifier ident, Dsymbols* decl)
+    {
+        super(decl);
+        this.ident = ident;
+    }
+
+    extern (D) this(Expression exp, Dsymbols* decl)
+    {
+        super(decl);
+        this.exp = exp;
+    }
+
+    extern (D) this(Identifier ident, Expression exp, Dsymbols* decl,
+                    CPPNamespaceDeclaration parent)
+    {
+        super(decl);
+        this.ident = ident;
+        this.exp = exp;
+        this.namespace = parent;
+    }
+
+    override Dsymbol syntaxCopy(Dsymbol s)
+    {
+        assert(!s);
+        return new CPPNamespaceDeclaration(
+            this.ident, this.exp, Dsymbol.arraySyntaxCopy(this.decl), this.namespace);
+    }
+
+    override Scope* newScope(Scope* sc)
+    {
+        auto scx = sc.copy();
+        scx.linkage = LINK.cpp;
+        scx.namespace = this;
+        return scx;
+    }
+
+    override const(char)* toChars() const
+    {
+        return toString().ptr;
+    }
+
+    extern(D) override const(char)[] toString() const
+    {
+        return "extern (C++, `namespace`)";
```

Comment:
Should include the identifier or expression instead of "`namespace`".

Reply:
`Nspace` doesn't do that (and it allocates)
I was considering doing a refactor where `toString` takes a sink, but it has been put off so far.